### PR TITLE
Enhacements.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

- Upload infection tests 100%.
- Upload phpunit tests 100%.
- Fixed more docs.
- Fix minor corrections.

### CODE REVIEW

File: src/NavBar.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ final class NavBar extends Widget
     private string $role = 'navigation';
 
     /**
-     * The ARIA label of the navbar.
+     * Returns a new instance with the specified aria-label attribute for the current element.
```
Comment (vjik) [inline src/NavBar.php:45]:
```suggestion
     * Returns a new instance with the specified `aria-label` attribute for the current element.
```

File: src/ProgressBar.php [hunk 2]
```diff
@@ -192,15 +193,16 @@ protected function run(): string
 
     private function build(array $attributes): array
     {
-        /** @var string */
-        $attributes['id'] ??= (Html::generateId($this->autoIdPrefix) . '-progressbar');
+        if (!array_key_exists('id', $attributes)) {
+            /** @var string */
+            $attributes['id'] = (Html::generateId($this->autoIdPrefix) . '-progressbar');
```
Comment (vjik) [inline src/ProgressBar.php:198]:
```suggestion
            $attributes['id'] = Html::generateId($this->autoIdPrefix) . '-progressbar';
```

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/NavBar.php (hunk 1 above)
Comment (terabytesoftw) [reply src/NavBar.php:45]: done.

File: src/ProgressBar.php (hunk 2 above)
Comment (terabytesoftw) [reply src/ProgressBar.php:198]: done.

**Review by vjik — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**